components/layout_2020/fragment_tree/box_fragment.rs

```diff
@@ -126,10 +124,14 @@ impl BoxFragment {
         // > value) a block-level or inline-level block container that is a scroll container
         // > always has a last baseline set, whose baselines all correspond to its block-end
         // > margin edge.
+        //
+        // This applies even if there is no baseline set.
+        let mut baselines = Baselines::default();
         if style.establishes_scroll_container() {
-            last_inflow_baseline_offset = Some(
-                content_rect.size.block + padding.block_end + border.block_end + margin.block_end,
-            );
+            baselines.last = Some(
+                (content_rect.size.block + padding.block_end + border.block_end + margin.block_end)
+                    .into(),
+            )
```

Comment:
I think it would be better to move this into `with_baselines()`.

Reply:
We aren't sure that `with_baselines()` will be called and we need to unconditionally set the baseline to the end position. I could move this to `with_baselines()` and call that always in the constructor, but it's a bit odd I think.

Reply:
I've improved this comment so it's a bit clearer why it's happening here instead of in `with_baselines()`.